code/game/machinery/doors/door.dm

```diff
@@ -49,13 +49,18 @@
 	//return
 
 /obj/machinery/door/Bumped(atom/AM)
-	if(operating || emagged) return
+	if(operating || emagged)
+		return
+
 	if(isliving(AM))
 		var/mob/living/M = AM
 		if(world.time - M.last_bumped <= 10) return	//Can bump-open one airlock per second. This is to prevent shock spam.
 		M.last_bumped = world.time
-		if(!M.restrained())
-			bumpopen(M)
+		if(M.restrained())
+			if(req_access_txt != "0")

```

Comment:
I'm not a big fan of this but it was the only way I found to check if the door is available to everyone...